src/redux/sagas/extensions/extensionEnable.ts

```diff
@@ -0,0 +1,84 @@
+import { all, call, put, takeEvery } from 'redux-saga/effects';
+
+import { ActionTypes } from '../../actionTypes';
+import { Action } from '../../types/actions';
+import { getTxChannel } from '../transactions';
+import {
+  modifyParams,
+  putError,
+  refreshEnabledExtension,
+  removeOldExtensionClients,
+  setupEnablingGroupTransactions,
+  takeFrom,
+} from '../utils';
+
+function* extensionEnable({
+  meta,
+  payload,
+  payload: {
+    colonyAddress,
+    extensionData: { extensionId, isInitialized, initializationParams },
+  },
+}: Action<ActionTypes.EXTENSION_ENABLE>) {
+  const initChannelName = `${meta.id}-initialise`;
+
+  yield removeOldExtensionClients(colonyAddress, extensionId);
+
+  const initChannel = yield call(getTxChannel, initChannelName);
+
+  try {
+    if (!isInitialized && initializationParams) {
+      const initParams = modifyParams(initializationParams, payload);
+
+      const {
+        channels,
+        transactionChannels,
+        transactionChannels: { initialise },
+        createGroupTransaction,
+      } = yield setupEnablingGroupTransactions(
```

Comment:
Just a note about the `call` pattern often used in sagas (here e.g. would be `call(setupEnablingGroupTransactions, ...args)`. 

In the event we ever want to test the sagas, it's designed with that in mind. It doesn't affect the functionality, but may be worth sticking to it a) for consistency and b) should we ever wish to test them.

[Source](https://redux-saga.js.org/docs/basics/DeclarativeEffects/)

Reply:
I'm not yet very familiar with redux-saga, but aren't we supposed to use `call` for functions returning Promises? 
If I understand correctly, it's still testable the way the docs describe (it yields, so the value can be accessed with `iterator.next().value`) but is a synchronous step.
I'll go ahead and merge this PR but if you can confirm your suggestion is still relevant for synchronous functions, I'll change it later

Reply:
@jakubcolony Sorry I thought `setupEnablingGroupTransactions` was async since it's called using `yield`. If it's synchronous presumably it could be called directly. That said, an effect may still be useful for synchronous functions, in the sense that you don't need to test whether the function itself returns the correct data, only that it's being called correctly with the correct arguments. Unit testing of the function itself presumably wouldn't happen as part of the saga test. But I'm really just thinking out loud. Feel free to leave as is.